test/python/transpiler/test_clifford_t_passmanager.py

```diff
@@ -0,0 +1,233 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test Clifford+T transpilation pipeline"""
+
+import numpy as np
+from ddt import ddt, data
+
+from qiskit.converters import circuit_to_dag
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import QFTGate, iqp, GraphStateGate
+from qiskit.transpiler.passes.utils import CheckGateDirection
+from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
+from qiskit.transpiler import CouplingMap
+from qiskit.providers.fake_provider import GenericBackendV2
+from qiskit.quantum_info import get_clifford_gate_names
+
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+
+@ddt
+class TestCliffordTPassManager(QiskitTestCase):
+    """Test Clifford+T transpilation pipeline."""
+
+    @data(0, 1, 2, 3)
+    def test_cliffords_1q(self, optimization_level):
+        """Clifford+T transpilation of a circuit with single-qubit Clifford gates."""
+        qc = QuantumCircuit(3)
+        qc.h(0)
+        qc.s(1)
+        qc.h(2)
+        qc.sdg(2)
+        qc.h(2)
+        qc.h(2)
+
+        basis_gates = ["cx", "s", "sdg", "h", "t", "tdg"]
+        pm = generate_preset_pass_manager(
+            basis_gates=basis_gates, optimization_level=optimization_level
+        )
+        transpiled = pm.run(qc)
+        self.assertLessEqual(set(transpiled.count_ops()), set(basis_gates))
+
+    @data(0, 1, 2, 3)
+    def test_complex_clifford(self, optimization_level):
+        """Clifford+T transpilation of a more complex Clifford circuit."""
+        qc = QuantumCircuit(3)
+        qc.h(0)
+        qc.s(0)
+        qc.cx(1, 0)
+        qc.cx(0, 1)
+        qc.s(1)
+        qc.sdg(0)
+        qc.cx(1, 0)
+
+        basis_gates = ["cx", "s", "sdg", "h", "t", "tdg"]
+        pm = generate_preset_pass_manager(
+            basis_gates=basis_gates, optimization_level=optimization_level
+        )
+        transpiled = pm.run(qc)
+        self.assertLessEqual(set(transpiled.count_ops()), set(basis_gates))
+
+    @data(0, 1, 2, 3)
+    def test_rx(self, optimization_level):
+        """Clifford+T transpilation of a circuit with a single-qubit rotation gate,
+        requiring the usage of the Solovay-Kitaev decomposition.
+        """
+        qc = QuantumCircuit(1)
+        qc.rx(0.8, 0)
```

Comment:
if the angle is a multiplication of pi/2, does it know to use only clifford gates?
if the angle is a multiplication of pi/4, does it know to use only clifford and t/tdg gates?

Reply:
For the pi/4 angle, if the basis set is Clifford+T, the flow by definition will only use Clifford+T/Tdg gates. But you are correct, the SK algorithm will be exact and produce a very short sequence of gates. I am adding a test for that. I don't want to add a similar test for pi/2 angle before we can guarantee (again using the future Clifford resynthesis plugin) that there will be only Clifford gates in the result.